src/background/tab-state.js

```diff
@@ -77,14 +79,14 @@ export default function TabState(initialState, onchange) {
   /**
    * Replaces the H state for all tabs.
    *
-   * @param {TabStateMap} newState - A dictionary mapping tab ID to tab state objects.
-   *                   The provided state will be merged with the default
-   *                   state for a tab.
+   * @param {Record<number, Partial<State>>} newState -
+   *   Map of tab ID to state. The state is merged with the default state for a tab.
    */
   this.load = function (newState) {
     /** @type {TabStateMap} */
     const newCurrentState = {};
-    Object.keys(newState).forEach(function (tabId) {
+    Object.keys(newState).forEach(function (tabIdStr) {
```

Comment:
```suggestion
    Object.keys(newState).forEach(tabIdStr => {
```

As part of this modernisation, we could run `eslint` to automatically convert to arrow functions, when possible.

Reply:
Yes. This was on my list to do at some point.